flowmachine/flowmachine/features/location/majority_location.py

```diff
@@ -0,0 +1,82 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# -*- coding: utf-8 -*-
+
+from typing import List
+
+from flowmachine.core import Query
+from flowmachine.core.mixins import GeoDataMixin
+from flowmachine.features.utilities.subscriber_locations import BaseLocation
+
+
+class MajorityLocation(BaseLocation, Query):
+    """
+    A class for producing a list of subscribers along with the location (derived from `spatial_unit')
+    that they visited more than half the time. Takes a query that includes a 'subscribers' column,
+    a 'spatial_unit' attribute and a column to be used as weighting for locations (`location_count` for example)
+
+    Parameters
+    ----------
+    subscriber_location_weights: Query
+        The query object containing subscribers, locations, and weights.
+    weight_column: str
+        The column, when summed, that will produce the count used to threshold the majority
+    include_unlocatable: bool default False
+        If `True`, returns every unique subscriber in the `subscriber_location_weights` query, with
+        the location column as `NULL` if no majority is reached.
+        If `False`, returns only subscribers that have achieved a majority location
+    """
+
+    def __init__(
+        self,
+        subscriber_location_weights: Query,
+        weight_column: str,
+        include_unlocatable: bool = False,
+    ):
+        if "subscriber" not in subscriber_location_weights.column_names:
+            raise ValueError("`subscriber` not in subscriber_location_weights query")
+        if weight_column not in subscriber_location_weights.column_names:
+            raise ValueError("weight_column must exist in subscriber_subset")
+        if not getattr(subscriber_location_weights, "spatial_unit"):
```

Comment:
More correct to use 'hasattr' here. I can't imagine we'd ever have a spatial unit that evaluates as falsy, but if we did it would mess up this check.